src/pixel-ping.coffee

```diff
@@ -29,22 +29,36 @@ serialize = ->
 
 # Reset the `store`.
 reset = ->
+  old_store = store
   store = {}
+  old_store
+
+# Merge the given `store` with the current one.
+merge = (new_store) ->
+  for key, hits of new_store
+    store[key] or= 0
+    store[key] +=  hits
 
 # Flushes the `store` to be saved by an external API. The contents of the store
 # are sent to the configured `endpoint` URL via HTTP POST. If no `endpoint` is
 # configured, this is a no-op.
 flush = ->
   log store
   return unless config.endpoint
+  on_error = (message) ->
+    if !config.discard
+      merge(old_store)
```

Comment:
It may not matter because of hoisting, but I'd really prefer to see `old_store = reset()` above the definition of `on_error`, where we reference `old_store`. The early reference to `old_store` makes me think that there's supposed to be an `old_store` global.

Also, I think `merge(old_store) unless config.discard` would read nicely.

Reply:
Fixed!